terraform/terraform.go

```diff
@@ -0,0 +1,227 @@
+	"strings"
+	"time"
+)
+
+type TFState struct {
+	Resources []Resource `json:"resources"`
+}
+
+type Resource struct {
+	Module    string     `json:"module"`
+	Mode      string     `json:"mode"`
+	Type      string     `json:"type"`
+	Name      string     `json:"name"`
+	Instances []Instance `json:"instances"`
+}
+
+type Instance struct {
+	Attributes Attributes `json:"attributes"`
+}
+
+type Attributes struct {
+	Arn string `json:"arn"`
+}
+
+func main() {
+	manager := plugin2.New()
+	manager.SetNonInteractiveView()
+	err := manager.StartServer()
+	if err != nil {
+		panic(err)
+	}
+	err = manager.StartPlugin("rds-instance")
+	if err != nil {
+		panic(err)
+	}
+	for i := 0; i < 100; i++ {
+		runningPlg := manager.GetPlugin("kaytu-io/plugin-aws")
+		if runningPlg != nil {
+			break
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+	runningPlg := manager.GetPlugin("kaytu-io/plugin-aws")
+	if runningPlg == nil {
+		panic("running plugin not found")
+	}
+	cfg, err := server.GetConfig()
+	if err != nil {
+		panic(err)
+	}
+
+	for _, rcmd := range runningPlg.Plugin.Config.Commands {
+		if rcmd.Name == "rds-instance" {
+			preferences.Update(rcmd.DefaultPreferences)
+
+			if rcmd.LoginRequired && cfg.AccessToken == "" {
+				// login
+				panic("please login")
+			}
+			break
+		}
+	}
+	err = runningPlg.Stream.Send(&golang.ServerMessage{
+		ServerMessage: &golang.ServerMessage_Start{
+			Start: &golang.StartProcess{
+				Command:          "rds-instance",
+				Flags:            nil,
+				KaytuAccessToken: cfg.AccessToken,
+			},
+		},
+	})
+	if err != nil {
+		panic(err)
+	}
+	jsonOutput, err := manager.NonInteractiveView.WaitAndReturnResults("json")
+	if err != nil {
+		panic(err)
+	}
+
+	var jsonObj struct {
+		Items []*golang.OptimizationItem
+	}
+	err = json.Unmarshal([]byte(jsonOutput), &jsonObj)
+	if err != nil {
+		panic(err)
+	}
+
+	recommendation := map[string]string{}
+	for _, item := range jsonObj.Items {
+		var recommendedInstanceSize string
+		for _, device := range item.Devices {
+			for _, property := range device.Properties {
+				if property.Key == "Instance Size" {
+					recommendedInstanceSize = property.Recommended
+				}
+			}
+		}
+		recommendation[item.Id] = recommendedInstanceSize
+	}
+
+	folder := "/home/saleh/projects/keibi/terraform-aws-rds/examples/complete-postgres/"
+	err = filepath.Walk(folder, func(path string, info fs.FileInfo, err error) error {
+		if info.IsDir() || !strings.HasSuffix(info.Name(), ".tf") {
+			return nil
+		}
+		if strings.Contains(path, ".terraform") {
+			return nil
+		}
+		if info.Name() != "main.tf" {
+			return nil
+		}
+
+		contentBytes, err := os.ReadFile(path)
+		if err != nil {
+			return err
+		}
+
+		file, diags := hclwrite.ParseConfig(contentBytes, "filename.tf", hcl2.InitialPos)
+		if diags.HasErrors() {
+			return fmt.Errorf("%s", diags.Error())
+		}
+
+		body := file.Body()
+		localVars := map[string]string{}
+		for _, block := range body.Blocks() {
+			if block.Type() == "locals" {
```

Comment:
I think it's better to use switch case here 
Also we need to support the resource itself ([aws_db_instance](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/db_instance) and [aws_rds_cluster](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/rds_cluster)